resolver/resolver.go

```diff
@@ -0,0 +1,63 @@
+package resolver
+
+import (
+	"fmt"
+
+	"github.com/vultisig/recipes/types"
+)
+
+type MagicResolver struct {
+	treasuryConfig map[string]map[string]string
+}
```

Comment:
Hmm not a huge fan of this - we should likely have multiple resolvers

So something like 

```go

type Resolver interface {
    Resolve(constant string, asset string) (string, error)
 }
```

And then have a TreasuryResolver, ThorchainResolver etc.